test/unit/ui_utils_spec.js

```diff
@@ -189,18 +189,16 @@ describe("ui_utils", function () {
         count++;
       });
       function domEventListener() {
-        done.fail("shall not dispatch DOM event.");
+        throw new Error("Shall not dispatch DOM event.");
```

Comment:
Have you checked that an `Error` being thrown *inside* of the event handler is enough to propagate it to the "global" level?

Reply:
Yes, I tested this by explicitly dispatching the event on the document and seeing that this exception is raised then.